test/framework/src/main/java/org/elasticsearch/cluster/project/TestProjectResolvers.java

```diff
@@ -102,12 +103,25 @@ public boolean supportsMultipleProjects() {
 
     /**
      * This method returns a ProjectResolver that gives back the specified project-id when its getProjectId method is called.
-     * It also assumes it is the only project in the cluster state and throws if that is not the case.
+     * The ProjectResolver can work with cluster state containing multiple projects and its supportsMultipleProjects returns true.
      */
     public static ProjectResolver singleProject(ProjectId projectId) {
         return singleProject(projectId, false);
     }
 
+    /**
+     * This method returns a ProjectResolver that returns a random unique project-id.
+     * It also assumes it is the only project in the cluster state and throws if that is not the case.
+     * In addition, the ProjectResolvers returns false for supportsMultipleProjects.
+     */
+    public static ProjectResolver singleProjectOnly() {
+        return singleProjectOnly(ESTestCase.randomUniqueProjectId());
+    }
```

Comment:
I'm not sure I get the goal of the implementation of this method. I understand that this method is mostly used as a placeholder, so we don't actually care about the project ID. However, by supplying a random project ID _and checking if it exists in the cluster state_, don't we always just return an exception if any of the methods are called? Wouldn't it make more sense to create a dedicated project resolver instance that just throws on every use? That would make the intent of this method/project resolver more clear.

Reply:
It does work for `getProjectId()`, `supportsMultipleProjects()`, `executeOnProject(...)` etc . I am not sure whether any of the existing usages rely on any of these. Probably not, I need to check.